lib/std/setutils.nim

```diff
@@ -21,20 +21,32 @@ import typetraits, macros
     ## The allowed types of a built-in set.
 ]#
 
-template toSet*(iter: untyped): untyped =
+template toSet*[T: set](iter: untyped, _: typedesc[T]): T =
```

Comment:
Why add a generic parameter *and* a `typedesc` parameter? Can't you write `s.toSet[:T]`? And what about situations where you don't need to explicitly say the type, since it can be inferred (or is Nim's type inference not good enough for that)?

Reply:
> Why add a generic parameter and a typedesc parameter?

this was covered in https://github.com/nim-lang/RFCs/issues/40; as @zah summarized in https://github.com/nim-lang/RFCs/issues/40#issuecomment-380272387 which explained the rationale:
1. Use generic types for generic value params, where the types will be inferred from the call-site.
2. Use typedesc for all params that must be explicitly specified by the user.

There are plenty of arguments for this; besides the ones in https://github.com/nim-lang/RFCs/issues/40#issuecomment-380272387 here are some more:
* `macro fn(T: typedesc): untyped` works whereas `macro fn[T]()` can't do anything useful
* explicit generic instantiation requires the uglier `s.toSet[:T]` syntax for MCS and fully qualified calls
* newer procs in stdlib (including this module, see `fullSet`) use typedesc for cases like this, also all of std/typetraits, jsonutils, options, enumutils ...